### ISSUE TITLE
RequireSelfReference: false positive for anonymous class

### ISSUE THREAD

**Issue by LastDragon-ru** [slevomat/coding-standard#1539]
```xml
<?xml version="1.0"?>
<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="./vendor/squizlabs/php_codesniffer/phpcs.xsd">
    <rule ref="SlevomatCodingStandard.Classes.RequireSelfReference"/>
</ruleset>
```

```php
<?php declare(strict_types = 1);

namespace tmp;

class Bug {
    public static function bug(): self {
        return new class() extends Bug {
            // empty
        };
    }
}
```

Actual:
```txt
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 7 | ERROR | [x] "self" for local reference is required.
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

But `new class() extends self {}` cannot be used because "PHP Fatal error:  Cannot use 'self' as class name, as it is reserved".

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/fde811815f176b80fe7f83c59bd36191fc8783bc
